sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -1,15 +1,169 @@
 #include "xmp_file_format.h"
 
+#include "xmp_stream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+
+using std::string;
+using std::vector;
+
 namespace mip {
 namespace file {
 
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
+const XMP_StringPtr kMsipLabelNamespace = "http://www.microsoft.com/msip/label/1.0/";
+const XMP_StringPtr kMsipLabelExtendedNamespace = "http://www.microsoft.com/msip/label/extended/1.0/";
+const XMP_StringPtr kLabelsArrayName = "labels";
+const XMP_StringPtr kExtendedStructName = "extended";
```

Comment:
remind me what is XMP_StringPtr really?

Reply:
const char *

---
In reply to: [136423275](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r136423275) [](ancestors = 136423275)